File: bundles/org.eclipse.vjet.vsf.vjo.kernel/src/org/eclipse/vjet/vsf/utils/Array.java
/*******************************************************************************
 * Copyright (c) 2012 eBay Inc. and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     eBay Inc. - initial API and implementation
 *******************************************************************************/
package org.eclipse.vjet.vsf.utils;

import org.eclipse.vjet.dsf.javatojs.anno.AJsProxy;
import org.eclipse.vjet.dsf.dap.proxy.NativeJsProxy;
import org.mozilla.mod.javascript.Scriptable;
import org.eclipse.vjet.dsf.javatojs.anno.AExclude;
import org.eclipse.vjet.dsf.dap.proxy.INativeJsFuncProxy;
import org.eclipse.vjet.dsf.dap.proxy.NativeJsFuncProxy;
import org.eclipse.vjet.dsf.dap.proxy.NativeJsTypeRef;
import org.eclipse.vjet.dsf.javatojs.anno.AJavaOnly;

//NativeJsProxy for vjo.dsf.utils.Array.js
@org.eclipse.vjet.dsf.resource.utils.CodeGen("NativeJsProxyGenerator")
@AJsProxy
public class Array extends NativeJsProxy {

    /** for framework use only */
    @AExclude
    public Array(Scriptable nativeObj){
        super(nativeObj);
    }

    /** internal use only */
    protected Array(Object ...args){
        super(args);
    }

    public Array() {
        super();
    }

    public static Object[] copy(Object[] poSrcArr) {
        return callStaticWithName("vjo.dsf.utils.Array", "copy", Object[].class, poSrcArr);
    }

    public static Object[] remove(Object[] poSrcArr, int piIndex, String psValue) {
        return callStaticWithName("vjo.dsf.utils.Array", "remove", Object[].class, poSrcArr, piIndex, psValue);
    }

    public static Object[] insert(Object[] poSrcArr, int piIndex, String psValue) {
        return callStaticWithName("vjo.dsf.utils.Array", "insert", Object[].class, poSrcArr, piIndex, psValue);
    }

    public static Object[] shift(Object[] poSrcArr, int piIndex) {
        return callStaticWithName("vjo.dsf.utils.Array", "shift", Object[].class, poSrcArr, piIndex);
    }

    public static boolean contains(Object[] poSrcArr, Object poObject) {
        return callStaticWithName("vjo.dsf.utils.Array", "contains", Boolean.class, poSrcArr, poObject);
    }

    @AJavaOnly
    public static final NativeJsTypeRef<Array> prototype = NativeJsTypeRef.get(Array.class);

    public static final INativeJsFuncProxy<NativeJsTypeRef<Array>> copy = NativeJsFuncProxy.create(prototype, "copy");

    public static final INativeJsFuncProxy<NativeJsTypeRef<Array>> remove = NativeJsFuncProxy.create(prototype, "remove");

    public static final INativeJsFuncProxy<NativeJsTypeRef<Array>> insert = NativeJsFuncProxy.create(prototype, "insert");

    public static final INativeJsFuncProxy<NativeJsTypeRef<Array>> shift = NativeJsFuncProxy.create(prototype, "shift");

    public static final INativeJsFuncProxy<NativeJsTypeRef<Array>> contains = NativeJsFuncProxy.create(prototype, "contains");
}

File: eclipse/tests/org.eclipse.vjet.eclipse.javatojs.core.tests/src/org/eclipse/vjet/eclipse/javatojs/core/tests/codegen/java2js/data/Array.js
/*******************************************************************************
 * Copyright (c) 2012 eBay Inc. and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     eBay Inc. - initial API and implementation
 *******************************************************************************/
/* @org.eclipse.vjet.dsf.resource.utils.CodeGen("VjoGenerator") */
vjo.ctype('org.eclipse.vjet.eclipse.javatojs.core.tests.codegen.java2js.data.Array') //< public
.needs('org.eclipse.vjet.vjo.java.lang.StringUtil')
.inherits('org.eclipse.vjet.eclipse.javatojs.core.tests.codegen.java2js.data.child.Fields')
.props({
    s_sa:null //< private String[]
})
.protos({
    m_avg:5, //< private long
    m_la:null, //< private long[]
    m_twoDimArray:null, //< private int[][]
    //> public constructs()
    constructs:function(){
        this.m_la=[1+this.m_avg,2,3];
        this.m_twoDimArray=[[1,2,3],[4,5,6]];
    },
    //> public void initializer()
    initializer:function(){
        var avg=10; //<long
        var la=[1+avg,2,3]; //<long[]
        var twoDimArray=[[1,2,3],[4,5,6]]; //<int[][]
        var prefix="Pre"; //<String
        var sa=[prefix+"a",prefix+"b",prefix+"c"]; //<String[]
    },
    //> public void creation()
    creation:function(){
        var avg=3; //<int
        var la=vjo.createArray(0, avg+3); //<long[]
        var sa=vjo.createArray(null, avg+3); //<String[]
        var la3=vjo.createArray(0, avg); //<long[][][]
        for (var i=0;i<avg;i++){
            la3[i]=vjo.createArray(0, i+2);
            for (var j=0;j<i+2;j++){
                la3[i][j]=vjo.createArray(0, j+2);
                for (var k=0;k<j+2;k++){
                    la3[i][j][k]=k;
                }
            }
        }
    },
    //> public void access()
    access:function(){
        var avg=2; //<int
        var sa2=vjo.createArray(null, avg); //<String[][]
        for (var i=0;i<avg;i++){
            sa2[i]=vjo.createArray(null, i+2);
            var line="Row "+org.eclipse.vjet.vjo.java.lang.StringUtil._valueOf(i)+": "; //<String
            for (var j=0;j<i+2;j++){
                sa2[i][j]=org.eclipse.vjet.vjo.java.lang.StringUtil._valueOf(j);
                line+=sa2[i][j]+", ";
            }
        }
    }
})
.inits(function(){
    this.vj$.Array.s_sa=["a","b","c"];
})
.endType();